pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -311,6 +324,183 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Schedules [DelegationAction::Revoke] for the delegator, towards all delegated collator.
+	/// The last fulfilled request causes the delegator to leave the set of delegators.
+	pub(crate) fn delegator_schedule_revoke_all(
+		delegator: T::AccountId,
+	) -> DispatchResultWithPostInfo {
+		let mut state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		let mut updated_scheduled_requests = vec![];
+		let now = <Round<T>>::get().current;
+		let when = now.saturating_add(T::LeaveDelegatorsDelay::get());
+
+		// pre-validate that not all delegations already have a Revoke request.
+		let mut revoke_count = 0;
+		for bond in &state.delegations.0 {
+			let collator = bond.owner.clone();
+			let scheduled_requests = <DelegationScheduledRequests<T>>::get(&collator);
+			let maybe_revoke = scheduled_requests.iter().find(|req| {
+				req.delegator == delegator && matches!(req.action, DelegationAction::Revoke(_))
+			});
+			if let Some(_) = maybe_revoke {
+				revoke_count += 1;
+			}
+		}
+		if revoke_count == state.delegations.0.len() {
```

Comment:
I am not clear why we do this check here. Why do we iterate satte.delegations twice? can we do the afore check in the for loop below?

Reply:
Currently the rationale is that `revoke_all` mimics the leave behavior and the delegator actually "leaves" after execution. We can make the behavior a `batch_revoke` if that makes more sense and @crystalin agrees. It might confuse the use though.